### ISSUE TITLE
Editor dimming should be a core workbench feature

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#191757]
It looks like today the editor dimming is done from the outside via explicit knowledge of CSS selectors which is prone to breaking easily. Every new editor would have to be adopted here. This should rather be an option that every editor has to support built-in and then this CSS patching should go away.

Refs:

https://github.com/microsoft/vscode/blob/35be9bf683eace09796e59d54f1f225bbc3a7866/src/vs/workbench/contrib/accessibility/browser/unfocusedViewDimmingContribution.ts#L38-L62

//cc @jrieken

**Comment by jrieken** [comment]
Could this be implemented on the level of workbench parts? I believe there is a pretty good abstraction for those and dimming could be a click-through (`pointer-events: none`) overlay on top of inactive parts

**Comment by Tyriar** [comment]
It was much simpler initially but it turns out that using filter on an editor instance can impact the contents like the hover's position.

@jrieken interesting idea, right now the only workbench part that is dimmed is the terminal [by design](https://github.com/microsoft/vscode/issues/191610#issuecomment-1697478907). This could be a nice solution to the iframe and above hover position problem though 👍

**Comment by Tyriar** [comment]
Merging into https://github.com/microsoft/vscode/issues/236576

/duplicate
